### ISSUE TITLE
RAW: Upgrade Darktable to v4.4+

### ISSUE THREAD

**Issue by lastzero** [photoprism/photoprism#3741]
**As a Darktable user who has many RAW images with sidecar XMP files, I would like PhotoPrism to include the latest Darktable version so that the JPEGs generated from the RAW images look exactly as specified in the XMP files.**

As suggested in Pull Request #3735 by https://github.com/qlyoung, there is a newer version of Darktable available for installation from [opensuse.org](https://software.opensuse.org/download.html?project=graphics:darktable&package=darktable). To ensure that we fully understand the implications and possible differences to the Ubuntu distribution packages, we should know:

- [ ] If these packages do not support certain hardware (like GPUs that depend on the distribution's driver support) or file types (such as Apple ProRAW DNG or HEIC) that might otherwise be supported by the distribution packages, e.g. due to patented codecs?

- [ ] Whether they download and/or link to other libraries, so that updating those libraries in the base distribution - e.g., after a [security vulnerability becomes known](https://floss.social/@photoprism/111064346355243107) - would not fix the issue in Darktable?

- [ ] What the minimum processor instruction set is for ARM64 and AMD64, e.g. whether AVX or AVX2 support is required in Intel/AMD CPUs?

- [ ] How often the packages are updated, especially in case major security vulnerabilities are found in Darktable (or other related packages additionally installed from opensuse.org)?

**Comment by qlyoung** [comment]
assign me pls

**Comment by per2jensen** [comment]
I have been building Darktable for myself for years on Ubuntu (https://github.com/per2jensen/dt-on-ubuntu)
I could help out with this one

**Comment by lastzero** [comment]
That might be an option for us too! Would be great if we can ship the latest Darktable version with our next release.

**Comment by lastzero** [comment]
#### Installation Size

Based on my preliminary research, installing the SuSE packages consumes about 100 MB of additional storage (uncompressed) in the Docker image if you install it with this command:

```
echo 'deb http://download.opensuse.org/repositories/graphics:/darktable/xUbuntu_23.04/ /' | sudo tee /etc/apt/sources.list.d/graphics:darktable.list
curl -fsSL https://download.opensuse.org/repositories/graphics:darktable/xUbuntu_23.04/Release.key | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/graphics_darktable.gpg > /dev/null
sudo apt update
sudo apt install darktable
```

*Maybe the additional storage requirements can be reduced if you manually remove some of the previously installed packages (needs to be tested).*

### Feature Support

The compile options are shown as follows by the `darktable --version` command:

| Feature | v4.2.1 | v4.4.2 |
|-------------|-----------|----------|
 | bit depth  |64 bit | 64bit |
|  build | normal | normal |
|  SSE2 optimization | enabled | enabled |
|  OpenMP support | enabled | enabled |
|  OpenCL support | enabled | enabled | 
|  Lua support |API version 9.0.0 |  **API version 9.1.0** |
|  Colord support | enabled | enabled |
|  gPhoto2 support | enabled | enabled |
|GraphicsMagick support | enabled | enabled |
|  ImageMagick support | disabled | disabled |
|  libavif support | enabled | **disabled** |
|libheif support | enabled | enabled |
| libjxl support|  enabled | enabled |
|  OpenJPEG support |  enabled | enabled |
|  OpenEXR support | enabled | enabled |
|  WebP support | enabled | enabled |

**Comment by lastzero** [comment]
@qlyoung My last commit allows you to install the latest darktable version by adding "darktable" to the list of actions in the PHOTOPRISM_INIT environment variable. Please note that this is a temporary solution that currently only works with Ubuntu 23.04 base images.

**Comment by per2jensen** [comment]
I need to set up a dev env and start understanding how DT is integrated into photoprism :-)

**Comment by lastzero** [comment]
@per2jensen We are using the `darktable-cli` command, so there's no special integration e.g. on the library level.

**Comment by lastzero** [comment]
An updated preview build is now available for testing the changes (see [above](#issuecomment-1762854399)):

- https://docs.photoprism.app/getting-started/updates/#development-preview

**Comment by qlyoung** [comment]
It works! Thank you!

**Comment by lastzero** [comment]
Our new base image has Darktable v4.4.2 pre-installed (without needing external packages):
- https://packages.ubuntu.com/search?keywords=darktable&searchon=names&suite=mantic&section=all

@qlyoung @per2jensen Shall we close this issue when our next stable version has been released (ships with v4.4+ as requested) and open a new issue for building Darktable from source, if possible?

**Comment by qlyoung** [comment]
Sounds great to me, thank you for pursuing this among all your other priorities.
